docs/decision-records/decision-record-rule-types.md

```diff
@@ -0,0 +1,77 @@
+# ADR for AZA Scanner Rule Types 
+* Date: 2021-04-05
+
+## Context and Problem Statement
+
+There are a wide range of issues and challenge when it comes to integrating with VNETs in Azure that may not be immediately apparent to the customer or CSE dev crews. This project aims to identify those issues and notify the user of ways to fix the problem. This document represents the pros and cons of the different ways to scan Azure Resources and the decision process for evaluating which scenerios to support. 
+
+## Decision Drivers
+
+* How does it help the customer?
+* What are the use cases?
+* What are the limitations?
+* Can it scan existing Resources as well as not yet created Resources?
+
+## Considered Options
+
+* ARM Templates
+* What if deployments
+* Bicep deployments
+* Terraform
+* Resource Graph
+
+## Decision Outcome
+
+Chosen options: 
+(Best option) - "ARM Templates", because it allows for existing and yet to be created infrasturce to be scanned for issues.     
+(Second Best option) - "Terraform", because not all customer use ARM templates and it would be valuable to have the ability to run the rules against a multi-cloud IaC.  
+(First Chosen option) - "Resource Graph", We thought this would be the best option, but after further review, not all the rules could be run against Resource Graph because it doesn't support all the settings needed to evaluate some rules.  
+
+### Priorities
+MVP  
+* Resource Graph Rule
+* ARM template (exported at Resource Group Scope)  
+
+Future 
+* Arm Template (user supplied, deployment scopes)
+* Bicep option can be accomplished through compiling down to ARM Templates, so not much would need be changed to add this functionality.
+* Terraform rules are a great option and should be pursued after completing previous options.   
+* What if option is probably not worth pursuing.   
+
+## Pros and Cons of the Options 
+
+### ARM Templates
+
+* Good, because Bicep deployments can be converted to ARM templates
+* Good, because existing infrastructor can be exported as an ARM template 
+* Good, because not yet deployed infrastucture can be scanned before deployment  
+* Bad, because there are no Typscript Types for Azure Resources and ARM templates
```

Comment:
This might not be strictly true - I'm not sure. But it would definitely be a pain to go import & maintain all of the libraries per-service, so the point stands